src/freenas/etc/grub.d/10_truenas_linux

```diff
@@ -314,13 +315,11 @@ for dataset in `echo $datasets`; do
   fi
 
   truenas_version=`echo "$dataset" | sed "s|$roots_dataset/||"`
-  kernel_version=`zfs get -H -o value truenas:kernel_version "$dataset" 2>/dev/null`
-  if [ "x$kernel_version" = "x" -o "x$kernel_version" = "x-" ] ; then
-    continue
-  fi
+  kernel_version_prod=`dpkg -s linux-image-truenas-production-amd64 | grep '^Source:' | sed 's|Source: linux-||' 2>/dev/null`
```

Comment:
We can't do that. If we generate this list from, e.g., 23.10, `dpkg` will return kernel version used in 23.10, which will break 22.12 entries

Reply:
Thanks, I will update this.